### PR TITLE
Man page

### PR DESCRIPTION
**Pull request by Andrew-H3**
Hey,
It's Andrew. I started working on the man page for pat. I'm not sure how it works with packaging, but for now I just put it in the debian directory. It's basically empty for now, but I'll work on it as I can and try to get most of it done soon.

### CODE REVIEW

**Comment by martinhpedersen** [comment]
Hi Andrew!

Thank you for this. Looking forward to your commits :)

I suggest that you commit often, and that we do a squash merge into `master` when you are ready. That way, it is easier to follow the changes and discuss them as we work through the various commands and options.

**Comment by Andrew-H3** [comment]
It seems that I cannot change the pull request from my develop branch to my master branch. If you want to contribute directly to master I may need to make a new pull request. I can do that now or later, however you wish. Unless you know of another way to go.

**Comment by martinhpedersen** [comment]
I'm sorry Andrew, merging into `develop` is what I meant. Sorry about that!

**Comment by Andrew-H3** [comment]
Okay, no problem. Also, I liked the idea of having multiple man pages and will plan on doing that.

File: debian/pat-configure.1 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+.TH PAT 1 "2017-09-04" "Linux" "Pat Configure"
+.SH NAME
+pat configure \- opens pats configuration file using nano
```
Comment (martinhpedersen) [inline debian/pat-configure.1]: > using nano

Although this is true on many Linux systems, it depends on your system configuration and environment (https://github.com/la5nta/pat/blob/master/main.go#L488).

The default fallback is in fact `vi`, but I would prefer something like "opens Pats configuration file using the system default editor".

File: debian/pat-configure.1 [hunk 2]
```diff
@@ -0,0 +1,17 @@
+.TH PAT 1 "2017-09-04" "Linux" "Pat Configure"
```
Comment (martinhpedersen) [inline debian/pat-configure.1]: Disclaimer: I have no idea about best practices regarding man pages.

Why does this mention Linux? My hope is that we can include these man pages for most platforms (including Linux, macOS and *BSD).

What do you think Andrew?

File: debian/pat-configure.1 (hunk 1 above)
Comment (Andrew-H3) [reply debian/pat-configure.1]: Okay, I'll change that.

File: debian/pat-configure.1 (hunk 2 above)
Comment (Andrew-H3) [reply debian/pat-configure.1]: Most of the man page formatting guides I read have "Linux" there just to show what platform the page is written for. If you want this to be cross platform I can just remove that with no problem.

**Comment by Andrew-H3** [comment]
How do you like the way I have the commands and options listed?

**Comment by martinhpedersen** [comment]
I think it looks really good 👍

File: debian/pat-configure.1 (hunk 2 above)
Comment (martinhpedersen) [reply debian/pat-configure.1]: It is the footer line right?

For `man git`, `man ls` and `man perl` they all show the project's name and version.

I don't think we should include the version in every man page, but maybe replace it with `Pat`? I want to emphasize that I really have no clue about this, so please do it at you see fit.

**Comment by anarcat** [comment]
anything holding up this merge?

**Comment by martinhpedersen** [comment]
We have to figure out how to package it properly for the .deb packages and macOS .pkg. I plan on looking into it soon.

And, as I understand it, @Andrew-H3 is still working on writing some pages for the other sub-commands(?).

**Comment by Andrew-H3** [comment]
Depending on when you want to merge I can try to finish up another one or two in time. I won't have time in the next few weeks to do more than that.

**Comment by anarcat** [comment]
On 2017-09-29 17:04:44, Martin Hebnes Pedersen wrote:
> We have to figure out how to package it properly for the .deb packages and macOS .pkg. I plan on looking into it soon.

for the .deb: trivial. just list them in debian/manpages, iirc. e.g. the
simplest possible debian/manpages is probably just "*.1" and that will
pick up any manpage in the root dir. :)

if you want to ship them for other packages, you may want to move the
manpages into man/*.1 and use that as a pattern in debian/manpages,
however.

> And, as I understand it, @Andrew-H3 is still working on writing some pages for the other sub-commands(?). 

i'd say just ship it and fix them later. ;)

a.

**Comment by martinhpedersen** [comment]
OK, I guess we should include them in the [v0.5.0](https://github.com/la5nta/pat/milestone/17) release then. I have some work ahead of me for that release, maybe four weeks from now or something like that.

@Andrew-H3 - If you want to add some more pages until then, that would be great. But as I have mentioned before, I do not want to but any pressure on your (or any other contributor for that matter) :)

@anarcat - thanks for pointing out how to include them in the debian package. 👍

**Comment by martinhpedersen** [comment]
@Andrew-H3 - I'm preparing to to merge these changes into my develop branch.

If you don't mind, I would like to get your full name and callsign added to the contributors list in README.md. If that is OK with you, could you please add yourself to the list in a commit pushed to this PR?

**Comment by Andrew-H3** [comment]
Good to go. I would enjoy doing some more in the future but right now i'm pretty busy. Thanks for working with me.

**Comment by martinhpedersen** [comment]
Thank you Andrew. It's been a pleasure working with you, and I sincerely appreciate your contribution. Hope we can work together again soon :-)

**Pull request merged**